library/spdm_responder_lib/libspdm_rsp_receive_send.c

```diff
@@ -864,7 +866,19 @@ libspdm_return_t libspdm_build_response(void *spdm_context, const uint32_t *sess
         }
     }
 
-    return LIBSPDM_STATUS_SUCCESS;
+    status = LIBSPDM_STATUS_SUCCESS;
+done:
+    if (spdm_request->request_response_code == SPDM_VENDOR_DEFINED_REQUEST) {
```

Comment:
I don't think this needs to be limited to a VDM. Any encrypted message should probably be cleared.